### ISSUE TITLE
Chiffrer les mots de passe ZdS dans le fichier de config

### ISSUE THREAD

**Issue by lypwig** [firm1/zest-writer#267]
Parce que bon, faut avouer que ça craint un peu. :-P

[Ça n'a pas l'air si compliqué à faire](http://stackoverflow.com/questions/4487525/encrypt-and-decrypt-a-string-in-java).

**Comment by WinXaito** [comment]
Bon en sois si on peut l'encrypter, n'importe quelle logiciel externe pourra le décrypter, mais c'est vrai que ce serait déjà beaucoup mieux.

Edit: Il faudrait voir pour garder en mémoire le cookie de la session du site, comme ça on ne conserve jamais le mot de passe de l'utilisateur !!

**Comment by lypwig** [comment]
> Edit: Il faudrait voir pour garder en mémoire le cookie de la session du site, comme ça on ne conserve jamais le mot de passe de l'utilisateur !!

Comprend pas, tu veux récupérer le cookie du site pour t'en servir pour faire la connexion ? Comment tu veux le décrypter ?

**Comment by WinXaito** [comment]
Sur le site, si tu coches "Restez connecter", ça te génère un cookie de session. Se cookie est valable un certain temps (Je dirais dans les 2-3 mois).

On peux donc stocker se cookie. Comme ça on peut se connecter sans mémoriser le mot de passe, l'utilisateur devra se reconnecter tout les deux trois mois. (Sauf suppression manuel du cookie), en gros faire un fonctionnement identique au fonctionnement de notre navigateur lorsque l'on accède au site.

**Comment by lypwig** [comment]
Je comprends toujours pas. :/ Ne m'explique pas comment fonctionne le site, explique-moi plutôt comme tu envisagerait le fonctionnement de ZestWriter. Je ne vois pas comme ce dernier peut envoyer des requêtes à ZdS sans stocker le mot de passe.

**Comment by WinXaito** [comment]
Hm je ne vois pas comment expliquer autrement, surtout que ce que je dis reste une idée vague. Il faut encore voir si c'est faisable, mais je ne vois pas pourquoi ça ne le serais pas.

Peut-être @firm1 peut t'expliquer un peu mieux que moi ? (Fin s'il a aussi compris :') )

**Comment by firm1** [comment]
Il me semble que ce que @WinXaito essait d'expliquer, c'est que : 
- si tu encrypte le mot de passe dans ZW et que tu le stocke dans un fichier, étant donné que ZW se connecte au site avec un (login/password) et non avec un token, il faudra a un moment donné que ZW decrypte le mot de passe qu'il a encrypté. Donc si ZW sait decrypter, n'importe qui peut aussi le decrypter. c'est ce qu'on peut qualifier de solution dégradée.
- pour bien faire, il faudrait que ZW, au lieu de stocker un login/password, stocke le cookie de l'utilisateur lorsque celui-ci arrive à se connecter la première fois. Le cookie a l'avantage d'être "non lisible". Bien sur le cookie expirera un jour, et si c'est le cas, il faudra que ZW redemande à l'utilisateur de rentrer à nouveau ses login/password pour generer un nouveau cookie valide.

Est-ce que tu comprend mieux @roipoussiere ?

**Comment by lypwig** [comment]
> il faudrait que ZW, au lieu de stocker un login/password, stocke le cookie de l'utilisateur lorsque celui-ci arrive à se connecter la première fois.

Sur le site ou sur ZW ?

> si c'est le cas, il faudra que ZW redemande à l'utilisateur de rentrer à nouveau ses login/password

Sur le site ou sur ZW ?

Enfin bref, il semble que je n'ai pas les compétences pour ça je vous laisse sur ce ticket :P

**Comment by firm1** [comment]
POur les deux questions, le cookie il faut le stocker sur ZW.

**Comment by lypwig** [comment]
Oui, ça j'ai bien compris, mes questions étaient : _se connecter la première fois sur le site ou sur ZW ?_ et _rentrer à nouveau ses login/password sur le site ou sur ZW ?_.

**Comment by firm1** [comment]
c'est bien se connecter la première fois sur ZW et rentrer a nouveau ses login/password sur ZW.

**Comment by lypwig** [comment]
On peut faire la connexion à l'API en lui donnant le cookie ?

**Comment by firm1** [comment]
Malheuresement ZW n'utilise pas l'API REST du site ici car elle est très
limitée pour l'instant. Mais pour repondre a ta question, le cookie suffit
pour réaliser les opérations que l'on souhaite.

Le jeu. 13 oct. 2016 11:07, Nathanaël Jourdane notifications@github.com a
écrit :
